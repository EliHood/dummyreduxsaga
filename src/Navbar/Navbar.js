import React from 'react';
import {
  BrowserRouter as Router, Route, Link, Switch,
} from 'react-router-dom';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Home from '../containers/home';

const Navbar = () => (
  <Router>
    <AppBar position="static">
      <Toolbar>
        <Grid justify="space-between" container>
          <Typography variant="h6" style={{ color: '#fff' }}>
                   Boilerplate
          </Typography>
          {/* if is authenticated, will render authlinks
                    if not will render guest links
                */}
          <Grid item>
            <Button align="right">
              <Link style={{ color: '#fff', textDecoration: 'none' }} underline="none" to="/">
                            Home
              </Link>

            </Button>
          </Grid>
        </Grid>
      </Toolbar>
    </AppBar>

    <Route exact path="/" component={Home} />

  </Router>
);
export default Navbar;
