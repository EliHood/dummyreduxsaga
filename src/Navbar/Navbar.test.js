import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import AppBar from '@material-ui/core/AppBar';
import Navbar from './Navbar';

describe('Should render <Navbar/> ', () => {
  it('should render navbar component', () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper.find(AppBar)).toHaveLength(1);
  });

  it('should snapshot navbar', () => {
    const wrapper = shallow(<Navbar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
