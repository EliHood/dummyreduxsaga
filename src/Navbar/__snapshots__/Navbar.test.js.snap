// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render <Navbar/>  should snapshot navbar 1`] = `
<BrowserRouter>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Grid))
        container={true}
        justify="space-between"
      >
        <WithStyles(ForwardRef(Typography))
          style={
            Object {
              "color": "#fff",
            }
          }
          variant="h6"
        >
          Boilerplate
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Button))
            align="right"
          >
            <Link
              style={
                Object {
                  "color": "#fff",
                  "textDecoration": "none",
                }
              }
              to="/"
              underline="none"
            >
              Home
            </Link>
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Home)",
        "type": [Function],
      }
    }
    exact={true}
    path="/"
  />
</BrowserRouter>
`;
